setuptools/command/build_ext.py

```diff
@@ -16,15 +16,11 @@
 except ImportError:
     _build_ext = _du_build_ext
 
-try:
-    # Python 2.7 or >=3.2
```

Comment:
Thanks for having a look.
Yes, you're right about the need for a unit test in the test suite.

Here the intent of the "original" author (before https://github.com/pypa/setuptools/commit/c04abca6 broke it by importing the newer module) was to import the global `distutils.sysconfig._config_vars` dictionary, not to get the values, but to temporarily set them before calling `distutils.sysconfig.customize_compiler`. This is only for darwin, where building dynamic load libraries requires special compiler flags.

The problem is the `customize_compiler` function uses the `_config_vars` which are defined in the `distutils.sysconfig` module, not the one from the new `sysconfig` module. Therefor, importing `_CONFIG_VARS` from the latter, and modifying them, does not have any effect on `customize_compiler`.

I'm not sure about the relationship between `sysconfig._CONFIG_VARS` and `distutils.sysconfig._config_vars`.